### PR TITLE
AVX-14857: Find controller by inst id if name tag changed

### PR DESCRIPTION
**Pull request by kzhangavtx**

### CODE REVIEW

File: aviatrix_ha.py [hunk 1]
```diff
@@ -98,14 +98,19 @@ def _lambda_handler(event, context):
...
-        )['Reservations'][0]['Instances'][0]
+        inst_id = os.environ.get('INST_ID')
+        try:
+            controller_instanceobj = client.describe_instances(
+                Filters=[
+                    {'Name': 'instance-state-name', 'Values': ['running']},
+                    {'Name': 'tag:Name', 'Values': [instance_name]}]
+            )['Reservations'][0]['Instances'][0]
+        except IndexError:
+            controller_instanceobj = client.describe_instances(
```
Comment (arvindsree) [inline aviatrix_ha.py:109]: Could we add a log here? 

Also, we should not check if inst_id  is None (During creation)

File: aviatrix_ha.py [hunk 2]
```diff
@@ -98,14 +98,19 @@ def _lambda_handler(event, context):
...
+                InstanceIds=[inst_id])['Reservations'][0]['Instances'][0]
     except Exception as err:
-        err_reason = "Can't find Controller instance with name tag %s. %s" % (instance_name,
-                                                                              str(err))
+        err_reason = ("Can't find Controller instance with name tag %s "
+                      "or inst id %s. %s" % (instance_name, inst_id, str(err)))
```
Comment (arvindsree) [inline aviatrix_ha.py:113]: This is the error message seen by the end user. We could retain the older message if inst_id is None

File: aviatrix_ha.py (hunk 1 above)
Comment (kzhangavtx) [reply aviatrix_ha.py]: Sure, added a log, and only check if inst_id exists

File: aviatrix_ha.py (hunk 2 above)
Comment (kzhangavtx) [reply aviatrix_ha.py]: Changed the error message

**Review by arvindsree — APPROVED** [review]

**Pull request merged**